docs/src/components/code-examples.js

```diff
@@ -1,9 +1,123 @@
 import React from 'react'
-import Code from './code'
+import styled, { css } from 'styled-components'
+import { StaticQuery, graphql } from 'gatsby'
+import SyntaxHighligher from './syntax-highlighter'
 
-export default props => (
-  <div>
-    <div>code examples for {props.subject}</div>
-    <Code language="python">lambda x: x + 1</Code>
-  </div>
+const linkCSS = css`
+  line-height: 1.25;
+  padding: 1.25rem 1.5rem;
+  border-bottom: ${props =>
+    props.active ? '0.35rem solid #007099' : '0.35rem solid #e0e0e0'};
+  text-decoration: none;
+  display: block;
+  color: #007099;
+
+  &:hover {
+    color: #000;
+    cursor: pointer;
+  }
+`
+
+const Tabs = styled.div`
+  margin-top: 1rem;
+  border-top: 1px solid #cfcfcf;
+`
+
+const TabsNav = styled.div`
+  border-bottom: 0.35rem solid #e0e0e0;
+  width: 100%;
+`
+
+const Ul = styled.ul`
+  margin: 0 0 -0.35rem;
+  padding: 0;
+  display: inline-block;
+  line-height: 1.5;
+  list-style-type: none;
+`
+
+const NavItem = styled.li`
+  margin: 0;
+  padding: 0;
+  display: inline-block;
+`
+
+const NavLink = styled.label`
+  ${linkCSS};
+`
+
+const TabContent = styled.div`
+  display: ${props => (props.active ? 'block' : 'none')};
+`
+
+export const filterTransformSortCodeSnippets = ({ codeSnippets, path }) => {
+  if (!codeSnippets) {
+    return []
+  }
+
+  return codeSnippets
+    .filter(n => n.node.relativePath.startsWith(path))
+    .map(n => n.node)
+    .sort((a, b) => a.name.localeCompare(b.name))
+}
+
+export class CodeExamplesComponent extends React.Component {
+  state = {
+    activeTab: 0,
+  }
+
+  handleClick = index => {
+    this.setState({ activeTab: index })
+  }
+
+  render = () => {
+    const codeSnippets = filterTransformSortCodeSnippets({
+      codeSnippets: this.props.data.allCodeSamples.edges,
+      path: this.props.path,
+    })
+    return (
+      <Tabs>
+        <TabsNav>
+          <Ul>
+            {codeSnippets.map((s, i) => (
+              <NavItem key={i}>
```

Comment:
@andrewoh531 can't find your previous comment regarding `key`. But, the value has to be unique across items that are `map`ped. E.g. if you do this:

```jsx
const arr = ["a", "b", "c"];
arr.map((v, i) => <span key={i}>{v}</span>)
```

Then if the values in array change, it won't necessarily re-render, e.g.:

```jsx
const arr = ["c", "a", "b"]
```

Because the `i`ndex is still `0`, `1` and `2`.

See https://reactjs.org/docs/lists-and-keys.html#keys

> The best way to pick a key is to use a string that uniquely identifies a list item among its siblings. Most often you would use IDs from your data as keys

And

> We don’t recommend using indexes for keys if the order of items may change.

Reply:
If you're unsure of a good value for the "id", let's chat about it, but I guess you're just trying to find out what is unique about the code snippet and use that.

Reply:
Thanks for the info